tests/common/mod.rs

```diff
@@ -61,9 +63,22 @@ pub fn for_each_block_from_disk(mut handler: impl FnMut(Block, ChainState)) {
 
         // Parse state
         let mut accounts: HashMap<Address, PlainAccount> = serde_json::from_reader(BufReader::new(
-            File::open(format!("blocks/{block_number}/state_for_execution.json")).unwrap(),
+            File::open(format!("blocks/{block_number}/pre_state.json")).unwrap(),
         ))
         .unwrap();
+
+        // Parse block hashes
+        let block_hashes: BlockHashes =
```

Comment:
Not very important but the Haskell way is...

```rust
let block_hashes: BlockHashes =
    File::open(format!("blocks/{block_number}/block_hashes.json"))
        .map(|file| {
            serde_json::from_reader::<_, HashMap<U256, B256, ahash::RandomState>>(
                BufReader::new(file),
            )
            .unwrap()
            .into()
        })
        .unwrap_or_default();
```

Reply:
Both the current way and the Haskell way are not looking satisfying to me. Let me try to make it cleaner.

Reply:
I think the Haskell way is fine.